Assets/Scripts/ObjectFollowMousePosition.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObjectFollowMousePosition : MonoBehaviour
+{
+    private Camera mainCamera;
+
+    private void Awake() { mainCamera = Camera.main; }
+
+    private void Update()
+    {
+        // 스크린 상의 마우스 좌표를 기준으로 월드 상의 좌표를 구함
+        Vector3 position = new Vector3(Input.mousePosition.x, Input.mousePosition.y);
+        transform.position = mainCamera.ScreenToWorldPoint(position);
+
+        transform.position = new Vector3(transform.position.x, transform.position.y, 0);
```

Comment:
아하 ㅋㅋ transform.position을 두번 set하는 것 같았는데 제가 코드를 잘못 읽었네요.. 

```suggestion
        Vector3 position = new Vector3(Input.mousePosition.x, Input.mousePosition.y);
        Vector3 worldPosition = mainCamera.ScreenToWorldPoint(position);
        transform.position = new Vector3(worldPosition.position.x, worldPosition.position.y, 0);
```
이런 방식으로 transform.position을 두번 활용하는 방법 대신 worldPosition을 활용하는 건 어떤가요?
더 알아보기 쉬울 것 같아요 ㅎㅎ

추가로 제가 스크린 상의 마우스 좌표를 기준으로 월드 상의 좌표를 구하는 방법에 대해서 찾아봤는데, 
Camera.main.nearClipPlane을 활용하더라고요

MapScene에서는 nearClipPlane이 0이어서 기존 코드가 가능한건 아닌지 확인 부탁드려요 ㅎㅎ

[how-to-convert-the-mouse-position-to-world-space-in-unity-2d-3d](https://gamedevbeginner.com/how-to-convert-the-mouse-position-to-world-space-in-unity-2d-3d/#screen_to_world_3d)

Reply:
완성하신 후에 마무리 하시면서 get하는 부분과 set하는 부분을 분리하는 건 어떨까요??

```csharp
private void Update()
{
    SetPositionToMouse();
}

private void SetPositionToMouse()
{
    transform.position = GetMouseWorldPosition();
}

private Vector3 GetMouseWorldPosition()
{
    //스크린 상의 마우스 좌표를 기준으로 월드 상의 좌표
    return mouseWorldPosition;
}
```

더 읽기 편하고 좋을 것 같아요!

Reply:
찍어보니까 nearClipPlane은 0.3이던데, 음.. 모르겠네요. 뭔가 2D는 굳이 안 해줘도 괜찮은 것 같기도하네요